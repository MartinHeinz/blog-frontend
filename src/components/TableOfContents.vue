<template>
    <div>
        <a :href="href">
        <div v-bind:class="setStyle">{{ label }}</div>
        <TableOfContents
            v-for="(node, index) in nodes"
            :nodes="node.nodes"
            :key="index"
            :depth="depth + 1"
            :label="node.label"
            :href="node.href">
        </TableOfContents>
        </a>
    </div>
</template>

<script>

export default {
    name: 'TableOfContents',
    props: ['label', 'nodes', 'depth', 'href'],
    computed: {
        setStyle() {
            return `toc-level-${this.depth}`;
        },
    },
};
</script>

<style>

</style>
