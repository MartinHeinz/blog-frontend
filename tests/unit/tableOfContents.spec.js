import { createLocalVue, mount } from '@vue/test-utils';
import TableOfContents from '@/components/TableOfContents.vue';

const tree = {
    nodes: [
        {
            label: 'item1',
            href: '/item1',
            nodes: [
                {
                    label: 'item1.1',
                    href: '/item1.1',
                },
                {
                    label: 'item1.2',
                    href: '/item1.2',
                    nodes: [
                        {
                            label: 'item1.2.1',
                            href: '/item1.2.1',
                        },
                    ],
                },
            ],
        },
        {
            label: 'item2',
            href: '/item2',
        },
    ],
};

describe('TableOfContents.vue', () => {
    it('renders correcly', () => {
        const localVue = createLocalVue();
        const wrapper = mount(TableOfContents, {
            localVue,
            propsData: {
                label: tree.label,
                nodes: tree.nodes,
                href: tree.href,
                depth: 1,
            },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });
});
