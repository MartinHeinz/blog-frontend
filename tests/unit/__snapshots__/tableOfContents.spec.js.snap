// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableOfContents.vue renders correcly 1`] = `
<div><a>
    <div class="toc-level-1"></div>
    <div><a href="/item1">
        <div class="toc-level-2">item1</div>
        <div><a href="/item1.1">
            <div class="toc-level-3">item1.1</div>
          </a></div>
        <div><a href="/item1.2">
            <div class="toc-level-3">item1.2</div>
            <div><a href="/item1.2.1">
                <div class="toc-level-4">item1.2.1</div>
              </a></div>
          </a></div>
      </a></div>
    <div><a href="/item2">
        <div class="toc-level-2">item2</div>
      </a></div>
  </a></div>
`;
